src/include/index/scan_optimizer.h

```diff
@@ -0,0 +1,604 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// index.h
+//
+// Identification: src/include/index/scan_optimizer.h
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <vector>
+
+#include "index/index.h"
+#include "common/exception.h"
+#include "common/logger.h"
+#include "common/pool.h"
+#include "catalog/schema.h"
+#include "catalog/manager.h"
+#include "storage/tuple.h"
+#include "common/printable.h"
+
+#include "index/index_util.h"
+
+namespace catalog {
+class Schema;
+}
+
+namespace storage {
+class Tuple;
+}
+
+namespace peloton {
+namespace index {
+
+/*
+ * class ConjunctionScanPredicate - Represents a series of AND-ed predicates
+ *
+ * This class does not store the predicate itself (they should be stored as
+ * vectors elsewhere)
+ *
+ * There are two corner cases with this class: full index scan and point query
+ *
+ * For full index scan, no key allocation is done, and both low key and high key
+ * should not be touched on all cases, including construction of bounds and
+ * value late binding. For point query, only the low key should be touched,
+ * and it is exactly the key for construction of lower bound and for value
+ * late binding
+ */
+class ConjunctionScanPredicate {
+ private:
+   
+  // This is the list holding indices of the upper bound and lower bound
+  // for the scan key. We only keep index in the corresponding Value[]
+  // since some values might not be able to bind when constructing the
+  // IndexScanPlan
+  //
+  // The length of this vector should equal the number of columns in
+  // index key
+  std::vector<std::pair<oid_t, oid_t>> value_index_list;

```

Comment:
Use underscore for Class Data Members. The same for all other members below. (https://google.github.io/styleguide/cppguide.html) Actually I know it was probably already there before you modified, but it would be nice if you can apply the same refactor to tuple schema and index metadata...